Thirdweb/Thirdweb.Wallets/SmartWallet/SmartWallet.cs

```diff
@@ -321,19 +321,66 @@ public async Task ForceDeploy()
     }
 
     /// <summary>
-    /// Verifies if a signature is valid for a message using EIP-1271.
+    /// Verifies if a signature is valid for a message using EIP-1271 or ERC-6492.
     /// </summary>
     /// <param name="message">The message to verify.</param>
     /// <param name="signature">The signature to verify.</param>
     /// <returns>True if the signature is valid, otherwise false.</returns>
     public async Task<bool> IsValidSignature(string message, string signature)
     {
-        try
+        var isCounterFactual = signature.EndsWith(Constants.ERC_6492_MAGIC_VALUE[2..]);
+
+        // ERC-6492
+        if (isCounterFactual)
         {
-            var magicValue = await ThirdwebContract.Read<byte[]>(this._accountContract, "isValidSignature", message.StringToHex(), signature.HexToBytes()).ConfigureAwait(false);
-            return magicValue.BytesToHex() == new byte[] { 0x16, 0x26, 0xba, 0x7e }.BytesToHex();
+            var erc6492Sig = new ABIEncode().DecodeEncodedComplexType<Erc6492Signature>(signature.HexToBytes().Take(signature.Length - 32).ToArray());
+            var multicall3 = await ThirdwebContract.Create(this.Client, Constants.MULTICALL3_ADDRESS, this._chainId).ConfigureAwait(false);
+            List<Multicall3_Result> result;
+            try
+            {
+                result = await multicall3
+                    .Read<List<Multicall3_Result>>(
+                        method: "aggregate3",
```

Comment:
How come you're calling multicall3 here? Was expecting to see a encoded call with the universal validator abi 

I guess that works too but then you have a dependency on multicall3 being deployed. 

Seems cleaner to do the deploy+ isvalidsig call in solidity which is what the universal validator does